store/sqlstore/tokens_store.go

```diff
@@ -39,57 +39,68 @@ func (s SqlTokenStore) Save(token *model.Token) error {
 	if err := token.IsValid(); err != nil {
 		return err
 	}
-
-	if err := s.GetMaster().Insert(token); err != nil {
+	query, args, err := s.getQueryBuilder().
+		Insert("Tokens").
+		Columns("Token", "CreateAt", "Type", "Extra").
+		Values(token.Token, token.CreateAt, token.Type, token.Extra).
+		ToSql()
+	if err != nil {
+		return errors.Wrap(err, "token_tosql")
+	}
+	if _, err := s.GetMasterX().Exec(query, args...); err != nil {
 		return errors.Wrap(err, "failed to save Token")
 	}
 	return nil
 }
 
 func (s SqlTokenStore) Delete(token string) error {
-	if _, err := s.GetMaster().Exec("DELETE FROM Tokens WHERE Token = :Token", map[string]interface{}{"Token": token}); err != nil {
+	if _, err := s.GetMasterX().Exec("DELETE FROM Tokens WHERE Token = ?", token); err != nil {
 		return errors.Wrapf(err, "failed to delete Token with value %s", token)
 	}
 	return nil
 }
 
 func (s SqlTokenStore) GetByToken(tokenString string) (*model.Token, error) {
-	token := &model.Token{}
+	var token model.Token
 
-	if err := s.GetReplica().SelectOne(token, "SELECT * FROM Tokens WHERE Token = :Token", map[string]interface{}{"Token": tokenString}); err != nil {
+	if err := s.GetReplicaX().Get(&token, "SELECT * FROM Tokens WHERE Token = ?", tokenString); err != nil {
 		if err == sql.ErrNoRows {
 			return nil, store.NewErrNotFound("Token", fmt.Sprintf("Token=%s", tokenString))
 		}
 
 		return nil, errors.Wrapf(err, "failed to get Token with value %s", tokenString)
 	}
 
-	return token, nil
+	return &token, nil
 }
 
 func (s SqlTokenStore) Cleanup() {
 	mlog.Debug("Cleaning up token store.")
 	deltime := model.GetMillis() - model.MaxTokenExipryTime
-	if _, err := s.GetMaster().Exec("DELETE FROM Tokens WHERE CreateAt < :DelTime", map[string]interface{}{"DelTime": deltime}); err != nil {
+	if _, err := s.GetMasterX().Exec("DELETE FROM Tokens WHERE CreateAt < ?", deltime); err != nil {
```

Comment:
@agnivade not related with this PR but thinking if the `deltime := model.GetMillis() - model.MaxTokenExipryTime` should be defined in the app layer?

Reply:
I agree. Looking at session_store.go/Cleanup and job_store.go/Cleanup, they pass the time from the upper layer. I can create an HW for this.